.github/workflows/nightly-rebase-upstream.yml

```diff
@@ -0,0 +1,70 @@
+jobs:
+  NightlyRebaseUpstream:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false  # run all jobs in the matrix even if one fails
+      matrix:
+        julia_branch:
+          - { name: "backports-release-1.9+RAI", upstream: "backports-release-1.9"}
+          - { name: "backports-release-1.10+RAI", upstream: "backports-release-1.10"}
+          - { name: "master+RAI", upstream: "master"}
+    steps:
+      - name: Checkout the repo and our branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.julia_branch.name }}
+          # Needed to make rebasing work (https://github.com/actions/checkout/issues/372)
+          fetch-depth: 0
+      - name: Attempt the rebase
+        id: attempt_the_rebase
+        run: |
+          git config --global user.email "julia-engineering@relational.ai"
+          git config --global user.name "RAI CI (GitHub Action Automation)"
+
+          git remote add upstream https://github.com/JuliaLang/julia
+          git fetch --quiet upstream ${{ matrix.julia_branch.upstream }}}
+          git rebase upstream/${{ matrix.julia_branch.upstream }}
+        continue-on-error: true
+      - name: On rebase success, push
+        if: steps.attempt_the_rebase.outcome == 'success'
+        run: |
+          git push --force origin ${{ matrix.julia_branch.name }} # force-push, because of the rebase.
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # --- Install Julia: Copied from CompatHelper ------------------------
+      - name: On rebase failure, install Julia nightly
+        if: steps.attempt_the_rebase.outcome != 'success'
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: 'nightly'  # use the nightly build to do the Pkg.update()
+          arch: ${{ runner.arch }}
+      # --------------------------------------------------------------------
+      - name: On rebase failure, checkout jira-incident action
+        if: steps.attempt_the_rebase.outcome != 'success' && steps.attempt_resolve_manifest.outcome != 'success'
+        id: checkout-jira-indicent-action
+        uses: actions/checkout@v3
+        with:
+          repository: RelationalAI/raicloud-deployment
+          ssh-key: ${{ secrets.DELVECI_SSH }}
+          path: deployment-repo
+      - name: On rebase failure, create Jira incident
+        if: steps.attempt_the_rebase.outcome != 'success' && steps.attempt_resolve_manifest.outcome != 'success'
+        id: create-jira-indicent
+        uses: ./deployment-repo/.github/actions/create-jira-incident
+        with:
+          affected_service: Engineering - Julia Upgrades - Medium Priority Only
+          summary: "Nightly rebase failed for RelationalAI/julia branch `${{ matrix.julia_branch.name }}`"
+          description: "Failed to rebase RelationalAI/julia branch `${{ matrix.julia_branch.name }}` branch on top of `${{ matrix.julia_branch.upstream}}`.\nPlease manually rebase ${{ matrix.julia_branch.name }}\n"
```

Comment:
should we have persistent open PRs for `v1.x+RAI` -> `v1.x`, so that we can easily see the diff and so also easily see the merge conflicts / any rebase mishaps in the GitHub UI? I think that'd be helpful. If we do, then i think it'd also be helpful to link the PR here too

Reply:
Like, open a PR from `backports-release-1.9+RAI` -> `backports-release-1.9`? Or `backports-release-1.9+RAI` -> `v1.9.3`? For the former, I think the PR would open upstream, which we wouldn't want. For the latter, it would show us the diff of commits that have been proposed in the next patch release + the patches we're carrying; is that what you're thinking? If so, I think that could be useful, yes.

Reply:
ah, right

> Like, open a PR from backports-release-1.9+RAI -> backports-release-1.9

i was thinking this, but i meant a PR on this repo (not upstream), but that's because i was forgetting we aren't also keeping `backports-release-1.9` up-to-date with upstream `backports-release-1.9`. i.e. the PR diff would just show our patches to v1.9

> Or backports-release-1.9+RAI -> v1.9.3? ...it would show us the diff of commits that have been proposed in the next patch release + the patches we're carrying

that does sound kinda useful too, now you mention it 😆 But i'm more worried about seeing our changes than i am what's already in the upcoming Julia release from upstream

Reply:
Ok, I opened PRs between `backports-release-1.9+RAI` and `v1.9.3+RAI` for now so we can see the total diff of: 1) incoming changes that are proposed for next patch release and 2) our own patches we're carrying for the branch. They're now linked in the GHA failure mode and should make it easy to compare diffs.